search.php

```diff
@@ -14,30 +14,33 @@
 /**
  * Planet4 - Search functionality.
  */
-if ( is_main_query() && is_search() ) {
-	if ( 'GET' === filter_input( INPUT_SERVER, 'REQUEST_METHOD' ) ) {
-		$selected_sort    = filter_input( INPUT_GET, 'orderby', FILTER_SANITIZE_STRING );
-		$selected_filters = $_GET['f'] ?? ''; // phpcs:ignore
-		$filters          = [];
+if ( ! is_main_query()
+	|| ! is_search()
+	|| 'GET' !== filter_input( INPUT_SERVER, 'REQUEST_METHOD' )
```

Comment:
I guess we can even remove these checks? I don't think it should get to `search.php` if these conditions aren't met. The [examples in the documentation](https://codex.wordpress.org/Creating_a_Search_Page) also don't include checks like this.

Reply:
Good point, but could we get spammed by post requests to the search page (assuming it would not be cached by cloudflare, which I don't know) ?

<sub>I read too much, the doc says to [not use search.php](https://codex.wordpress.org/Creating_a_Search_Page#Creating_a_Search_Page_Template) as a filename :fearful: </sub>

Reply:
It's not cached at all by CF. So valid concern that this might help mitigate attacks. Though it would be unintentionally doing so, I don't see anything in the code indicating this was the intention.

Reply:
Your second small point is actually a big one. Our search indeed currently does what the documentation warns against and as a result it would execute a first query unused.

We actually did a workaround to prevent this query without majorly rewriting the code. Somehow [I added that to the plugin](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/1940bf58bf36180ac5d7af69fa497921ffc0110e/planet4-gutenberg-blocks.php#L310-L343) :facepalm:

Reply:
Created a ticket for potential follow-up on this https://jira.greenpeace.org/browse/PLANET-6541